diff --git a/target/linux/bmips/dts/bcm3383-technicolor-tc7200u.dts b/target/linux/bmips/dts/bcm3383-technicolor-tc7200u.dts
index c462055b88..704e213982 100644
--- a/target/linux/bmips/dts/bcm3383-technicolor-tc7200u.dts
+++ b/target/linux/bmips/dts/bcm3383-technicolor-tc7200u.dts
@@ -1,3 +1,16 @@
+&{/ubus} {
+	/delete-node/ serial@14e00520;
+
+	tc7200_uart0: serial@14e00500 {
+		compatible = "brcm,bcm6345-uart";
+		reg = <0x14e00500 0x18>;
+		interrupt-parent = <&periph_intc>;
+		interrupts = <2>;
+		clocks = <&periph_clk>;
+		status = "okay";
+	};
+};
+
 // SPDX-License-Identifier: GPL-2.0
 /dts-v1/;
 
@@ -7,19 +20,16 @@
 	compatible = "technicolor,tc7200u", "brcm,bcm3384-viper", "brcm,bcm33843-viper";
 	model = "Technicolor TC7200.U";
 
+	aliases {
+		serial0 = &tc7200_uart0;
+	};
+
 	chosen {
 		bootargs = "console=ttyS0,115200 earlycon";
-		stdout-path = &uart0;
+		stdout-path = "serial0:115200n8";
 	};
 };
 
-&uart0 {
-	reg = <0x14e00500 0x18>;
-	interrupt-parent = <&periph_intc>;
-	interrupts = <2>;
-	clocks = <&periph_clk>;
-	status = "okay";
-};
 /* TC7200.U SPI NOR probe. Disabled after invalid boot/panic. RAM-boot only. */
 &{/ubus} {
         spi: spi@14e01000 {
