commit 3fe5f69e32081fa88c8a93379a4bc57196eb40f5
Author: Joseph C. Lehner <joseph.c.lehner@gmail.com>
Date:   Sun Mar 5 11:05:29 2017 +0100

    bmips: add Technicolor TC7200 support

diff --git a/arch/mips/bmips/Kconfig b/arch/mips/bmips/Kconfig
index 2d60f25403de..83502575bb98 100644
--- a/arch/mips/bmips/Kconfig
+++ b/arch/mips/bmips/Kconfig
@@ -77,6 +77,10 @@ config DT_SFR_NEUFBOX6_SERCOMM
 	bool "SFR Neufbox 6 (Sercomm)"
 	select BUILTIN_DTB
 
+config DT_TECHNICOLOR_TC7200_VIPER
+	bool "Technicolor TC7200 (Viper)"
+	select BUILTIN_DTB
+
 endchoice
 
 endif
diff --git a/arch/mips/boot/dts/brcm/Makefile b/arch/mips/boot/dts/brcm/Makefile
index d61bc2aebf69..18232a42ad14 100644
--- a/arch/mips/boot/dts/brcm/Makefile
+++ b/arch/mips/boot/dts/brcm/Makefile
@@ -14,12 +14,14 @@ dtb-$(CONFIG_DT_COMTREND_VR3032U)	+= bcm63268-comtrend-vr-3032u.dtb
 dtb-$(CONFIG_DT_NETGEAR_CVG834G)	+= bcm3368-netgear-cvg834g.dtb
 dtb-$(CONFIG_DT_SFR_NEUFBOX4_SERCOMM)	+= bcm6358-neufbox4-sercomm.dtb
 dtb-$(CONFIG_DT_SFR_NEUFBOX6_SERCOMM)	+= bcm6362-neufbox6-sercomm.dtb
+dtb-$(CONFIG_DT_TECHNICOLOR_TC7200_VIPER) += bcm3383_viper_technicolor_tc7200.dtb
 
 dtb-$(CONFIG_DT_NONE) += \
 	bcm3368-netgear-cvg834g.dtb \
 	bcm6358-neufbox4-sercomm.dtb \
 	bcm6362-neufbox6-sercomm.dtb \
 	bcm63268-comtrend-vr-3032u.dtb \
+	bcm3383_viper_technicolor_tc7200.dtb \
 	bcm93384wvg.dtb \
 	bcm93384wvg_viper.dtb \
 	bcm96358nb4ser.dtb \
diff --git a/arch/mips/boot/dts/brcm/bcm3383_viper.dtsi b/arch/mips/boot/dts/brcm/bcm3383_viper.dtsi
new file mode 100644
index 000000000000..02e3aa405636
--- /dev/null
+++ b/arch/mips/boot/dts/brcm/bcm3383_viper.dtsi
@@ -0,0 +1,110 @@
+/ {
+	#address-cells = <1>;
+	#size-cells = <1>;
+	compatible = "brcm,bcm3383-viper";
+
+	memory@0 {
+		device_type = "memory";
+
+		/* Typical ranges.  The bootloader should fill these in. */
+		reg = <0 0>;
+	};
+
+	cpus {
+		#address-cells = <1>;
+		#size-cells = <0>;
+
+		/* 1/2 of the CPU core clock (standard MIPS behavior) */
+		mips-hpt-frequency = <300000000>;
+
+		cpu@0 {
+			compatible = "brcm,bmips4350";
+			device_type = "cpu";
+			reg = <0>;
+		};
+	};
+
+	cpu_intc: interrupt-controller {
+		#address-cells = <0>;
+		compatible = "mti,cpu-interrupt-controller";
+
+		interrupt-controller;
+		#interrupt-cells = <1>;
+	};
+
+	clocks {
+		periph_clk: periph_clk {
+			compatible = "fixed-clock";
+			#clock-cells = <0>;
+			clock-frequency = <50000000>;
+		};
+	};
+
+	aliases {
+		uart0 = &uart0;
+	};
+
+	ubus {
+		#address-cells = <1>;
+		#size-cells = <1>;
+
+		compatible = "brcm,ubus", "simple-bus";
+		ranges;
+		/* No dma-ranges on Viper. */
+
+		periph_intc: interrupt-controller@14e00048 {
+			compatible = "brcm,bcm3380-l2-intc";
+			reg = <0x14e00048 0x4 0x14e0004c 0x4>;
+			interrupt-controller;
+			#interrupt-cells = <1>;
+			interrupt-parent = <&cpu_intc>;
+			interrupts = <4>;
+		};
+
+		uart0: serial@14e00500 {
+			compatible = "brcm,bcm6345-uart";
+			reg = <0x14e00500 0x18>;
+			interrupt-parent = <&periph_intc>;
+			interrupts = <2>;
+			clocks = <&periph_clk>;
+			status = "disabled";
+		};
+
+		ehci: usb@12e00200 {
+			compatible = "brcm,bcm3384-ehci", "generic-ehci";
+			reg = <0x12e00200 0x100>;
+			big-endian;
+			interrupt-parent = <&periph_intc>;
+			interrupts = <24>;
+			needs-reset-on-resume;
+			status = "disabled";
+		};
+
+		spi: spi@14e01000 {
+			compatible = "brcm,bcm6328-hsspi";
+			reg = <0x14e01000 0x100>;
+
+			#address-cells = <1>;
+			#size-cells = <1>;
+
+			interrupt-parent = <&periph_intc>;
+			interrupts = <7>;
+
+			status = "disabled";
+		};
+
+		nand: nand@14e02200 {
+			compatible = "brcm,brcmnand", "brcm,brcmnand-v4.0";
+			reg-names = "nand", "nand-cache";
+			reg = <0x14e02200 0xf4>, <0x14e02600 0x1000>;
+
+			#address-cells = <1>;
+			#size-cells = <0>;
+
+			interrupt-parent = <&periph_intc>;
+			interrupts = <31>;
+
+			status = "disabled";
+		};
+	};
+};
diff --git a/arch/mips/boot/dts/brcm/bcm3383_viper_technicolor_tc7200.dts b/arch/mips/boot/dts/brcm/bcm3383_viper_technicolor_tc7200.dts
new file mode 100644
index 000000000000..b50f1a463548
--- /dev/null
+++ b/arch/mips/boot/dts/brcm/bcm3383_viper_technicolor_tc7200.dts
@@ -0,0 +1,43 @@
+/dts-v1/;
+
+/include/ "bcm3383_viper.dtsi"
+
+/ {
+	compatible = "technicolor,tc7200-viper", "brcm,bcm3383-viper";
+	model = "Technicolor TC7200 (Viper)";
+
+	memory@0 {
+		device_type = "memory";
+		reg = <0x00000000 0x08000000>;
+	};
+
+	chosen {
+		bootargs = "console=ttyS0,115200";
+		stdout-path = &uart0;
+	};
+};
+
+&uart0 {
+	status = "okay";
+};
+
+&ehci {
+	status = "okay";
+};
+
+&spi {
+	status = "okay";
+	
+	mp25p80@0 {
+		#address-cells = <1>;
+		#size-cells = <1>;
+		compatible = "mx25l8008e";
+		spi-max-frequency = <10000000>;
+		reg = <0 0>;
+	};
+};
+
+&nand {
+	status = "okay";
+};
+
